src/reim/autoPhaseCorrection.js

```diff
@@ -0,0 +1,230 @@
+      reTmp.push(re[i]);
+      imTmp.push(im[i]);
+      i++
+    } 
+    // for (; finalPeaks[i] && i < length; i++) {
+      
+    // }
+    if (reTmp.length > minRegSize) {
+      res.push(autoPhaseRegion(reTmp, imTmp, x0));
+    }
+  }
+
+  // TODO: Still some corrections needed. In the paper they remove the outlayers interatively
+  // until they can perform a regression witout bad points. Can someone help here?
+  let [ph1, ph0] = weightedLinearRegression(
+    res.map((r) => r.x0 / length),
+    res.map((r) => r.ph0),
+    res.map((r) => r.area / 1e11),
+  );
+
+  let phased = phaseCorrection(
+    { re, im },
+    (ph0 * Math.PI) / 180,
+    (ph1 * Math.PI) / 180,
+  );
+
+  return { data: phased, ph0, ph1 };
+}
+
+function autoPhaseRegion(re, im, x0) {
+  let start = -180;
+  let stop = 180;
+  let nSteps = 20;
+  let maxSteps = 3;
+  let bestAng = 0;
+  while (maxSteps > 0) {
+    let dAng = (stop - start) / (nSteps + 1);
+    let minArea = Number.MAX_SAFE_INTEGER;
+    bestAng = start;
+    for (let i = start; i <= stop; i += dAng) {
+      let phased = phaseCorrection({re, im}, (Math.PI * i) / 180, 0);
+      let negArea = 0;
+      for (let j = 0; j < re.length; j++) {
+        if (phased.re[j] < 0) {
+          negArea += -phased.re[j];
+        }
+      }
+      if (negArea < minArea) {
+        minArea = negArea;
+        bestAng = i;
+      }
+    }
+    start = bestAng - dAng;
+    stop = bestAng + dAng;
+    maxSteps--;
+  }
+
+  // Calculate the area for the best angle
+  let phased = phaseCorrection({re, im}, (Math.PI * bestAng) / 180, 0);
+  let area = 0;
+  let sumX = 0;
+  for (let j = 0; j < re.length; j++) {
+    area += phased.re[j];
+    sumX += phased.re[j] * (j + x0);
+  }
+
+  return { ph0: bestAng, area, x0: sumX / area };
+}
+
+function holoborodko(s) {
+  let dk = new Array(s.length);
+  for (let i = 5; i < s.length - 5; i++) {
+    dk[i] =
+      (42 * (s[i + 1] - s[i - 1]) +
+        48 * (s[i + 2] - s[i - 2]) +
+        27 * (s[i + 3] + s[i - 3]) +
+        8 * (s[i + 4] - s[i - 4]) +
+        s[i + 5] -
+        s[i - 5]) /
+      512;
+  }
+  //Fill the borders
+  for (let i = 0; i < 5; i++) {
+    dk[i] = dk[5];
+    dk[s.length - i - 1] = dk[s.length - 6];
+  }
+
+  return dk;
+}
+
+function robustBaseLineRegionsDetection(s, minRegSize) {
+  let mask = new Array(s.length);
+  for (let i = 0; i < s.length; i++) {
+    mask[i] = false;
+  }
+
+  //Recursivelly check for points greater than 3 times the sdt
+  let change = true;
+  while (change) {
+    let res = stats(s, mask);
+    let noiseLevel = 3 * res.std;
+    let mean = res.mean;
+    change = false;
+    for (let i = 0; i < s.length; i++) {
+      if (Math.abs(s[i] - mean) > noiseLevel && !mask[i]) {
+        change = true;
+        mask[i] = true;
+      }
+    }
+  }
+
+  // Clean up mask by merging peaks blocks, separated by just a few points(4??).
+  let count = 0;
+  let prev = 0;
+  const SMALL = minRegSize;
```

Comment:
We reserve uppercase variable for real constants that are usually defined on the top of the file like Math.PI. It should be a normal lowercase variable in this case.